### PR TITLE
Add stock search functionality

### PR DESCRIPTION
**Pull request by Cebix90**

### CODE REVIEW

File: src/main/java/com/cebix/investmenttrackerapp/services/StockService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @Autowired
+    private StocksAPIHandler stocksAPIHandler;
+
+    public Stock getStockData(String ticker, String date) throws Exception {
+        if (date == null || date.isEmpty()) {
+            date = LocalDate.now().minusDays(1).toString();
+        }
+
+        try {
+            LocalDate fromDate = LocalDate.parse(date);
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/services/StockService.java:29]: Parsing the same date

File: src/main/java/com/cebix/investmenttrackerapp/services/StockService.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            if (fromDate.isAfter(toDate)) {
+                throw new DateOrderException("The parameter 'to' cannot be a time that occurs before 'from'");
+            }
+
+            if (fromDate.isAfter(currentDate) || fromDate.equals(currentDate)) {
+                throw new FutureDateException("You can only select a date at least one day in the past.");
+            }
+        } catch (DateTimeParseException e) {
+            throw new InvalidDateException("Invalid date format");
```
Comment (blazejprzyluski) [inline src/main/java/com/cebix/investmenttrackerapp/services/StockService.java:41]: Redundant exception

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -2,4 +2,7 @@ spring.application.name=InvestmentTrackerApp
 spring.datasource.url=jdbc:postgresql://localhost:5432/postgres
 spring.datasource.username=postgres
 spring.datasource.password=mateusz
-hibernate.dialect=org.hibernate.dialect.PostgresPlusDialect
\ No newline at end of file
+hibernate.dialect=org.hibernate.dialect.PostgresPlusDialect
+
+spring.thymeleaf.prefix=classpath:/templates/
```
Comment (blazejprzyluski) [inline src/main/resources/application.properties:7]: redundant, to be deleted

**Review by blazejprzyluski — COMMENTED** [review]
Please add tests where applicable

File: src/main/java/com/cebix/investmenttrackerapp/services/StockService.java (hunk 1 above)
Comment (Cebix90) [reply src/main/java/com/cebix/investmenttrackerapp/services/StockService.java:29]: needed corrections made

File: src/main/java/com/cebix/investmenttrackerapp/services/StockService.java (hunk 2 above)
Comment (Cebix90) [reply src/main/java/com/cebix/investmenttrackerapp/services/StockService.java:41]: corrected

**Review by blazejprzyluski — APPROVED** [review]
Approved!

**Pull request merged**